packages/atlas-service/src/main.ts

```diff
@@ -369,33 +393,36 @@ export class AtlasService {
       'Start atlas service token refresh'
     );
     try {
-      if (waitForOIDCPluginStateUpdateEvents) {
-        // We expect only one promise below to resolve, to clean up listeners
-        // that never fired we are setting up an abort controller
-        const listenerController = new AbortController();
-        try {
-          // When oidc-plugin starts internal token refresh, we will wait
-          // first for the auth flow to either succeed to fail and then
-          // proceed by requesting token once again from the plugin to sync
-          // token back to the atlas service state
-          await Promise.race([
-            // We are not using `refresh-succeeded` / `refresh-failed` events
-            // here because those happen BEFORE `allowedFlows` method is called
-            // (see oidc-plugin flow diagram) meaning that we need to wait until
-            // the whole request token flow went through to make sure that sign
-            // in flow is not allowed while we are refreshing the token
-            once(this.oidcPluginLogger, 'mongodb-oidc-plugin:auth-succeeded', {
-              signal: listenerController.signal,
-            }),
-            once(this.oidcPluginLogger, 'mongodb-oidc-plugin:auth-failed', {
+      // We expect only one promise below to resolve, to clean up listeners
+      // that never fired we are setting up an abort controller
+      const listenerController = new AbortController();
+      try {
+        // When oidc-plugin starts internal token refresh with the tryRefresh
+        // flow, we wait for a certain event sequence before we can safely
+        // request new token from the plugin
```

Comment:
[not blocking for this PR] I think part of what's confusing for me is what "safely" means. What's unsafe about requesting a token while a background token refresh is in progress? I get that Compass is not supposed to start a new browser flow, but I don't understand why we specifically wait for background refreshes to finish.